barbarik.py

```diff
@@ -834,11 +829,7 @@ def one_experiment(self, experiment, j, i, numExperiments, tj):
                 i += 1
                 ok, breakExperiment = exp.one_experiment(experiment, j, i, numExperiments, tj)
 
-                if ok is None:
```

Comment:
Wait, `None, None` is a valid return! Why is this being removed?

```
        if self.thresholdSolutions < self.minSamples:
            return None, None
```

Reply:
If ok is None or if ok is False, the correct thing to do is to continue to the next iteration. The real problem was with the 
`i -= 1` on line 841. The decrement is both theoretically unnecessary AND leads to an infinite loop as we don't change the value of seed before running the experiment again.